airflow/timetables/datasets.py

```diff
@@ -0,0 +1,90 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import collections.abc
+import typing
+
+import attrs
+
+from airflow.datasets import Dataset
+from airflow.exceptions import AirflowTimetableInvalid
+from airflow.timetables.simple import DatasetTriggeredTimetable
+from airflow.utils.types import DagRunType
+
+if typing.TYPE_CHECKING:
+    import pendulum
+
+    from airflow.timetables.base import DagRunInfo, DataInterval, TimeRestriction, Timetable
+
+
+class DatasetTimetable(DatasetTriggeredTimetable):
+    """Combine time-based scheduling with event-based scheduling."""
+
+    def __init__(self, time: Timetable, event: collections.abc.Collection[Dataset]) -> None:
```

Comment:
I am also wondering here if the second param `event` here might be confused with the `EventsTimetable` that we have 
https://github.com/apache/airflow/blob/c01daf811925816e9ae09b78c37b9ff8d87ce691/airflow/timetables/events.py#L33

Reply:
Should we just rename this to `dataset` instead?

Reply:
Or `datasets`

Reply:
Changed it here: https://github.com/apache/airflow/pull/36710/commits/9c3abb7b2e2dad2fd7122db535a364b2433087c1